gittestserver/server.go

```diff
@@ -164,10 +182,21 @@ func (s *GitServer) Root() string {
 	return s.config.Dir
 }
 
-// HTTPAddress returns the address of the HTTP git server.
+// HTTPAddress returns the address of the HTTP git server. This
+// includes credentials, if s.Auth() was called, so it will work
+// transparently.
 func (s *GitServer) HTTPAddress() string {
 	if s.httpServer != nil {
-		return s.httpServer.URL
+		u, err := url.Parse(s.httpServer.URL)
+		if err != nil {
+			panic(err)
+		}
+		if s.password != "" {
+			u.User = url.UserPassword(s.username, s.password)
+		} else if s.username != "" {
+			u.User = url.User(s.username)
+		}
+		return u.String()
```

Comment:
Doesn't this complicate situations where you want to e.g. mock a `GitRepository` resource including the `Secret` required to authenticate, and do not want to include the credential part of the URL in the URL you define (as those are defined in the secret)?

If so, maybe it would be wise to add a `RepositoryHTTPAddress(repository string)` (and maybe also the `SSH` version) that does not include any credentials for HTTP and only the user for SSH?

Reply:
Yeahp, that's the other branch of that decision. Good suggestion -- in fact perhaps I should have `HTTPAddressWithCredentials()` and adjust _my_ tests to use that.

Reply:
Let me rework this a little.